aerogear/ups-config-operator
# add service catalog definitions and implement binding delete

Adds the definitions for ServiceCatalog CRDs to the config operator and implements a lookup & delete function for service bindings.

This depends on https://github.com/aerogearcatalog/unifiedpush-apb/pull/52

@darahayes you can use this to delete a binding (and trigger unbind) when your code detects that a variant in UPS is missing.

@aliok This now also contains all the changes required for the UI.

Verification steps (UI part):

1. Needs to be used together with https://github.com/aerogearcatalog/unifiedpush-apb/pull/55
1. Deploy the APB
1. Deploy a mobile client apb (any client is ok, ios, android, cordova or xamarin)
1. Build this branch of the config operator locally (`make docker_build`)
1. Update the ups deploymentconfig and set the tag for the ups-config-operator to `latest` (this will make Openshift pick up the local image that you built in the previous step)
1. Create an Android binding
1. Verify that everything worked, variant created etc.
1. In Openshift, go to `Resources -> Other Resources`
1. From the dropdown pick `Mobile Clients`
1. Select the mobile client and view it's yaml: you should see an annotation in the form of `org.aerogear.binding.localregistry-unifiedpush-apb-<random>/variant-<platform>`
1. The annotation should have a URL attached to it. Open the URL.
1. It should directly open the variant in UPS.
1. Now delete the binding
1. The annotation should be removed from the mobile client.
1. Try the same thing with an iOS binding.
